lib/autoproj/cli/daemon.rb

```diff
@@ -101,67 +101,98 @@ def clear_and_dump_cache
             # @return [void]
-            def handle_pull_request_push(pull_request)
+            def handle_pull_request_modifications(pull_request, events)
+                # Check whether the pull request closed, and if we're aware
+                unless pull_request.open?
+                    if cache.include?(pull_request)
+                        handle_pull_request_closed(pull_request)
+                    end
+                    return
+                end
+
+                unless cache.include?(pull_request)
+                    return handle_pull_request_opened(pull_request)
+                end
+
+                handle_pull_request_changes(pull_request, events)
+            end
+
+            # Handle changes made to an existing, already open, pull request
+            def handle_pull_request_changes(pull_request, _events)
                 overrides = buildconf_manager.overrides_for_pull_request(pull_request)
                 return unless cache.changed?(pull_request, overrides)
 
                 branch_name = branch_name_by_pull_request(pull_request)
-
                 buildconf_manager.commit_and_push_overrides(branch_name, overrides)
                 cache.add(pull_request, overrides)
 
                 Autoproj.message "Push detected on #{pull_request.base_owner}/"\
                     "#{pull_request.base_name}##{pull_request.number}"
 
-                bb.build_pull_request(pull_request)
+                bb.post_pull_request_changes(pull_request)
                 cache.dump
             end
 
-            # @param [Github::PushEvent] push_event
-            # @return [void]
-            def handle_push_event(push_event, mainline: nil, pull_request: nil)
-                return handle_mainline_push(push_event) if mainline
-                return if update_failed? || buildconf_pull_request?(pull_request)
-
-                handle_pull_request_push(pull_request)
-            end
-
-            # @param [Github::PullRequest] pull_request
-            # @return [void]
+            # Handle a pull request that has just been opened
             def handle_pull_request_opened(pull_request)
                 branch_name = branch_name_by_pull_request(pull_request)
-
                 overrides = buildconf_manager.overrides_for_pull_request(pull_request)
-                return unless cache.changed?(pull_request, overrides)
 
                 Autoproj.message "Creating branch #{branch_name} "\
                     "on #{buildconf_package.owner}/#{buildconf_package.name}"
 
                 buildconf_manager.commit_and_push_overrides(branch_name, overrides)
-                bb.build_pull_request(pull_request)
+                bb.post_pull_request_changes(pull_request)
 
                 cache.add(pull_request, overrides)
                 cache.dump
             end
 
-            # @param [Github::PullRequest] pull_request
-            # @return [void]
+            # Handle a pull request that has just been closed
```

Comment:
Will do.

(I also think that if you're spending the time to write the annotations, it would be very good to actually document the method. Like saying what it's meant to do)